src/filemap.rs

```diff
@@ -36,16 +36,95 @@ pub fn write_all_files(file_map: &FileMap,
                        config: &Config)
                        -> Result<(HashMap<String, String>), io::Error> {
     let mut result = HashMap::new();
+    output_heading(mode).ok();
+
     for filename in file_map.keys() {
         let one_result = try!(write_file(&file_map[filename], filename, mode, config));
         if let Some(r) = one_result {
             result.insert(filename.clone(), r);
         }
     }
+    output_trailing(mode).ok();
 
     Ok(result)
 }
 
+pub fn output_heading(mode: WriteMode) -> Result<(), io::Error> {
+    let stdout = stdout();
+    let mut stdout = stdout.lock();
+    match mode {
+        WriteMode::Checkstyle => {
+            let mut xml_heading = String::new();
+            xml_heading.push_str("<?xml version=\"1.0\" encoding=\"utf-8\"?>");
+            xml_heading.push_str("\n");
+            xml_heading.push_str("<checkstyle version=\"4.3\">");
+            try!(write!(stdout, "{}", xml_heading));
+            Ok(())
+        }
+        _ => Ok(()),
```

Comment:
I wasn't able to get get `if let` working. I kept getting errors like

```
src/checkstyle.rs:18:8: 18:12 error: binary operation `==` cannot be applied to type `WriteMode` [E0369]
src/checkstyle.rs:18     if mode == WriteMode::Checkstyle {
```

And I wasn't sure if implementing PartialEq was wanted.